internal/vsp/feepayment.go

```diff
@@ -25,38 +25,24 @@ import (
 	"github.com/decred/vspd/types/v2"
 )
 
-var prng lockedRand
-
-type lockedRand struct {
-	mu   sync.Mutex
-	rand *uniformprng.Source
-}
-
-func (r *lockedRand) int63n(n int64) int64 {
-	r.mu.Lock()
-	defer r.mu.Unlock()
-	return r.rand.Int63n(n)
-}
-
-// duration returns a random time.Duration in [0,d) with uniform distribution.
-func (r *lockedRand) duration(d time.Duration) time.Duration {
-	return time.Duration(r.int63n(int64(d)))
+// randomInt64 returns a random int64 in [0,n).
+func randomInt64(n int64) int64 {
+	i, err := rand.Int(rand.Reader, big.NewInt(n))
```

Comment:
I suggest keeping the existing logic even if we aren't importing the internal/uniformprng package anymore.

rand.Int is meant to generate large random numbers within a prime field, such as when generating private keys.  It is overkill for this use case.

Here's a version that reads from crypto/rand directly instead of the PRNG:

```Go
// randInt63 returns a cryptographically random 63-bit positive integer as an
// int64.
func randInt63() int64 {
	buf := make([]byte, 8)
	_, err := rand.Read(buf)
	if err != nil {
		panic(fmt.Sprintf("unhandled crypto/rand error: %v", err))
	}
	return int64(binary.LittleEndian.Uint64(buf) &^ (1 << 63))
}

// randInt63n returns, as an int64, a cryptographically-random 63-bit positive
// integer in [0,n) without modulo bias.
// It panics if n <= 0.
func randInt63n(n int64) int64 {
	if n <= 0 {
		panic("invalid argument to int63n")
	}
	n--
	mask := int64(^uint64(0) >> bits.LeadingZeros64(uint64(n)))
	for {
		v := randInt63() & mask
		if v <= n {
			return v
		}
	}
}
```

Reply:
Done